zfec/filefec.py

```diff
@@ -251,7 +253,7 @@ def decode_from_files(outf, infiles, verbose=False):
         m = nm
         if not (k is None or k == nk):
             raise CorruptedShareFilesError("Share files were corrupted -- share file %r said that k was %s but another share file previously said that k was %s" % (f.name, nk, k,))
-        if k > len(infiles):
+        if not (k is None or k <= len(infiles)):
```

Comment:
The extra check is needed for Python 3 because `int` and `None` can't be compared anymore (unorderable types).